pymeasure/instruments/attocube/adapters.py

```diff
@@ -0,0 +1,109 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import time
+
+from pymeasure.adapters import TelnetAdapter
+
+
+class AttocubeConsoleAdapter(TelnetAdapter):
+    """ Adapter class for connecting to the Attocube Standard Console. This
+    console is a Telnet prompt with password authentication.
+
+    :param host: host address of the instrument
+    :param port: TCPIP port
+    :param passwd: password required to open the connection
+    :param kwargs: Any valid key-word argument for TelnetAdapter
+    """
+    def __init__(self, host, port, passwd, **kwargs):
+        self.read_termination = '\r\n'
+        self.write_termination = self.read_termination
+        super().__init__(host, port, **kwargs)
+        time.sleep(self.query_delay)
+        super().read()  # clear messages sent upon opening the connection
+        # send password and check authorization
+        self.write(passwd, checkAck=False)
+        time.sleep(self.query_delay)
+        ret = super().read()
+        authmsg = ret.split(self.read_termination)[1]
+        if authmsg != 'Authorization success':
+            raise Exception("Attocube authorization failed ('%s')" % authmsg)
+        # switch console echo off
+        _ = self.ask('echo off')
+
+    def check_acknowledgement(self, reply, msg=""):
+        """ checks the last reply of the instrument to be 'OK', otherwise a
+        ValueError is raised.
+
+        :param reply: last reply string of the instrument
+        :param msg: optional message for the eventual error
+        """
+        if reply != 'OK':
+            if msg == "":  # clear buffer
+                msg = reply
+                super().read()
+            raise ValueError("AttocubeConsoleAdapter: Error after command '%s'"
+                             " with message '%s'" % (self.lastcommand, msg))
+
+    def read(self):
+        """ Reads a reply of the instrument which consists of two or more
+        lines. The first ones are the reply to the command while the last one
+        is 'OK' or 'ERROR' to indicate any problem. In case the reply is not OK
+        a ValueError is raised.
+
+        :returns: String ASCII response of the instrument.
+        """
+        raw = super().read().strip(self.read_termination)
+        # one would want to use self.read_termination as 'sep' below, but this
+        # is not possible because of a firmware bug resulting in inconsistent
+        # line endings
```

Comment:
Do you mean that using '\n' as sep results in one of the strings (i think ret) will often have a carriage return ('\r')? In that case you could consider adding a `ret = ret.strip()` to remove this for a more consistent output.

Reply:
exactly, will change that. The extra '\r' is not harmful but better to remove it